### PR TITLE
feat(useDraggable): add stopPropagation option

### LINKED ISSUE #1386: useDraggable    没有阻止元素冒泡   在某些情况下会造成bug
### Describe the bug

useDraggable    没有阻止元素冒泡   在某些情况下会造成bug

### Reproduction

e.stopPropagation();

### System Info

```Shell
e.stopPropagation();
```

### Used Package Manager

npm

### Validations

- [X] Follow our [Code of Conduct](https://github.com/vueuse/vueuse/blob/main/CODE_OF_CONDUCT.md)
- [X] Read the [Contributing Guidelines](https://github.com/vueuse/vueuse/blob/main/CONTRIBUTING.md).
- [X] Read the [docs](https://vueuse.org/guide).
- [X] Check that there isn't [already an issue](https://github.com/vueuse/vueuse/issues) that reports the same bug to avoid creating a duplicate.
- [X] Make sure this is a VueUse issue and not a framework-specific issue. For example, if it's a Vue SFC related bug, it should likely be reported to https://github.com/vuejs/core instead.
- [X] Check that this is a concrete bug. For Q&A open a [GitHub Discussion](https://github.com/vueuse/vueuse/discussions).
- [X] The provided reproduction is a [minimal reproducible example](https://stackoverflow.com/help/minimal-reproducible-example) of the bug.

### PR DESCRIPTION
**Pull request by webfansplz**
### Description

Close #1386 

### Additional context

---

### What is the purpose of this pull request? 

- [ ] Bug fix
- [x] New Feature
- [ ] Documentation update
- [ ] Other

### Before submitting the PR, please make sure you do the following

- [x] Read the [Contributing Guidelines](https://github.com/vueuse/vueuse/blob/main/CONTRIBUTING.md).
- [x] Read the [Pull Request Guidelines](https://github.com/vueuse/vueuse/blob/main/packages/guidelines.md).
- [x] Check that there isn't already a PR that solves the problem the same way to avoid creating a duplicate.
- [x] Provide a description in this PR that addresses **what** the PR is solving, or reference the issue that it solves (e.g. `fixes #123`).
- [ ] Ideally, include relevant tests that fail without this PR but pass with it.

### CODE REVIEW

File: packages/core/useDraggable/index.ts [hunk 1]
```diff
@@ -79,6 +86,12 @@ export function useDraggable(target: MaybeRef<HTMLElement | SVGElement | null>,
     if (unref(options.preventDefault))
       e.preventDefault()
   }
+
+  const stopPropagation = (e: PointerEvent) => {
+    if (unref(options.stopPropagation))
+      e.stopPropagation()
+  }
```
Comment (antfu) [inline packages/core/useDraggable/index.ts:93]: We can merge this into the `preventDefault` and rename it as `handleEvent`

**Comment by canyonwan** [comment]
`DOM: 
          <div class="fixed z-50 cursor-move bg-e-tools-box-icon bg-no-repeat bg-full h-20 w-20" ref="toolsBoxRef" :style="style" @click="openModal()">tools box</div>
 
  Method: 
        const { width, height } = useWindowSize()

        const { style } = useDraggable(toolsBoxRef, {
          initialValue: { x: width.value - 120, y: height.value - 120 },
          stopPropagation: true
        })`
Problem : 
           When I move him, when I release the mouse, triggered the click method, but i already set ` stopPropagation: true` 
           option.
ENV:
       "vite": "^2.9.8",
       "vue": "^3.2.33",
       "@vueuse/core": "^8.4.2",

**Comment by Xdy1579883916** [comment]
````html
<component
    :is="draggable ? UseDraggable : 'div'"
    class="logo"
    :initialValue="{ top: '50%', right: 0 }"
    :preventDefault="true"
    :stopPropagation="true"
  >
    <img
      width="120"
      draggable="false"
      src="xxxx.png"
      @click.stop.prevent="$emit('open')"
    />
  </component>
````

我使用usedraggable组件，当我释放鼠标时，也触发了click方法，但我已经设置了： `：stopPropagation=“true”`；
我读了源代码，可能是以下问题造成的：

I use the usedraggable component， when I release the mouse, triggered the click method too, but i already set : 
`:stopPropagation="true"`; I read the source code, which may be caused by the following problems：

image

**Comment by canyonwan** [comment]
> ```
> <component
>     :is="draggable ? UseDraggable : 'div'"
>     class="logo"
>     :initialValue="{ top: '50%', right: 0 }"
>     :preventDefault="true"
>     :stopPropagation="true"
>   >
>            width="120"
>       draggable="false"
>       src="xxxx.png"
>       @click.stop.prevent="$emit('open')"
>     />
>   </component>
> ```
> 
> 我使用usedraggable组件，当我释放鼠标时，也触发了click方法，但我已经设置了： `：stopPropagation=“true”`； 我读了源代码，可能是以下问题造成的：
> 
> I use the usedraggable component， when I release the mouse, triggered the click method too, but i already set : `:stopPropagation="true"`; I read the source code, which may be caused by the following problems：
> 
> image
所以现在还要等他们修复是吗

**Comment by webfansplz** [comment]
> > ```
> > <component
> >     :is="draggable ? UseDraggable : 'div'"
> >     class="logo"
> >     :initialValue="{ top: '50%', right: 0 }"
> >     :preventDefault="true"
> >     :stopPropagation="true"
> >   >
> >      >       width="120"
> >       draggable="false"
> >       src="xxxx.png"
> >       @click.stop.prevent="$emit('open')"
> >     />
> >   </component>
> > ```
> > 
> > 
> >     
> >       
> >     
> > 
> >       
> >     
> > 
> >     
> >   
> > 我使用usedraggable组件，当我释放鼠标时，也触发了click方法，但我已经设置了： `：stopPropagation=“true”`； 我读了源代码，可能是以下问题造成的：
> > I use the usedraggable component， when I release the mouse, triggered the click method too, but i already set : `:stopPropagation="true"`; I read the source code, which may be caused by the following problems：
> > image
> > 所以现在还要等他们修复是吗

PR已经合并,等发版哈

**Comment by canyonwan** [comment]
> > > ```
> > > <component
> > >     :is="draggable ? UseDraggable : 'div'"
> > >     class="logo"
> > >     :initialValue="{ top: '50%', right: 0 }"
> > >     :preventDefault="true"
> > >     :stopPropagation="true"
> > >   >
> > >      > >       width="120"
> > >       draggable="false"
> > >       src="xxxx.png"
> > >       @click.stop.prevent="$emit('open')"
> > >     />
> > >   </component>
> > > ```
> > > 
> > > 
> > >     
> > >       
> > >     
> > > 
> > >       
> > >     
> > > 
> > >     
> > >   
> > > 我使用usedraggable组件，当我释放鼠标时，也触发了click方法，但我已经设置了： `：stopPropagation=“true”`； 我读了源代码，可能是以下问题造成的：
> > > I use the usedraggable component， when I release the mouse, triggered the click method too, but i already set : `:stopPropagation="true"`; I read the source code, which may be caused by the following problems：
> > > image
> > > 所以现在还要等他们修复是吗
> 
> PR已经合并,等发版哈

我已经将vueuse升级到了8.6.0  但是我这个还是不行

**Comment by canyonwan** [comment]
> > > ```
> > > <component
> > >     :is="draggable ? UseDraggable : 'div'"
> > >     class="logo"
> > >     :initialValue="{ top: '50%', right: 0 }"
> > >     :preventDefault="true"
> > >     :stopPropagation="true"
> > >   >
> > >      > >       width="120"
> > >       draggable="false"
> > >       src="xxxx.png"
> > >       @click.stop.prevent="$emit('open')"
> > >     />
> > >   </component>
> > > ```
> > > 
> > > 
> > >     
> > >       
> > >     
> > > 
> > >       
> > >     
> > > 
> > >     
> > >   
> > > 我使用usedraggable组件，当我释放鼠标时，也触发了click方法，但我已经设置了： `：stopPropagation=“true”`； 我读了源代码，可能是以下问题造成的：
> > > I use the usedraggable component， when I release the mouse, triggered the click method too, but i already set : `:stopPropagation="true"`; I read the source code, which may be caused by the following problems：
> > > image
> > > 所以现在还要等他们修复是吗
> 
> PR已经合并,等发版哈

我已经将vueuse升级到了8.6.0  但是我这个还是不行

**Comment by canyonwan** [comment]
> > > ```
> > > <component
> > >     :is="draggable ? UseDraggable : 'div'"
> > >     class="logo"
> > >     :initialValue="{ top: '50%', right: 0 }"
> > >     :preventDefault="true"
> > >     :stopPropagation="true"
> > >   >
> > >      > >       width="120"
> > >       draggable="false"
> > >       src="xxxx.png"
> > >       @click.stop.prevent="$emit('open')"
> > >     />
> > >   </component>
> > > ```
> > > 
> > > 
> > >     
> > >       
> > >     
> > > 
> > >       
> > >     
> > > 
> > >     
> > >   
> > > 我使用usedraggable组件，当我释放鼠标时，也触发了click方法，但我已经设置了： `：stopPropagation=“true”`； 我读了源代码，可能是以下问题造成的：
> > > I use the usedraggable component， when I release the mouse, triggered the click method too, but i already set : `:stopPropagation="true"`; I read the source code, which may be caused by the following problems：
> > > image
> > > 所以现在还要等他们修复是吗
> 
> PR已经合并,等发版哈

我已经将vueuse升级到了8.6.0  但是我这个还是不行 不过我用的不是组件的形式 而是像这样https://github.com/vueuse/vueuse/pull/1391#issuecomment-1129658414

**Comment by canyonwan** [comment]
@webfansplz 

> DOM: `<div class="fixed z-50 cursor-move bg-e-tools-box-icon bg-no-repeat bg-full h-20 w-20" ref="toolsBoxRef" :style="style" @click="openModal()">tools box</div>`
> 
> Method:
> 
> ```
>     const { width, height } = useWindowSize()
> 
>     const { style } = useDraggable(toolsBoxRef, {
>       initialValue: { x: width.value - 120, y: height.value - 120 },
>       stopPropagation: true
>     })`
> ```
> 
> ENV: "vite": "^2.9.8", "vue": "^3.2.33", "@vueuse/core": "^8.4.2",
> 
> **Problem : When I move him, when I release the mouse, triggered the click method, but i already set ` stopPropagation: true` option.**

@webfansplz  像我这样还是不行

**Comment by canyonwan** [comment]
> ```
> <component
>     :is="draggable ? UseDraggable : 'div'"
>     class="logo"
>     :initialValue="{ top: '50%', right: 0 }"
>     :preventDefault="true"
>     :stopPropagation="true"
>   >
>            width="120"
>       draggable="false"
>       src="xxxx.png"
>       @click.stop.prevent="$emit('open')"
>     />
>   </component>
> ```
> 
> 我使用usedraggable组件，当我释放鼠标
WeChat6f20e7ccf8c5289bde7c7bdf5b9ed8c4
时，也触发了click方法，但我已经设置了： `：stopPropagation=“true”`； 我读了源代码，可能是以下问题造成的：
> 
> I use the usedraggable component， when I release the mouse, triggered the click method too, but i already set : `:stopPropagation="true"`; I read the source code, which may be caused by the following problems：
> 

> image

请问你是怎么引入的UseDraggable组件的  我一直

**Comment by canyonwan** [comment]
> ```
> <component
>     :is="draggable ? UseDraggable : 'div'"
>     class="logo"
>     :initialValue="{ top: '50%', right: 0 }"
>     :preventDefault="true"
>     :stopPropagation="true"
>   >
>            width="120"
>       draggable="false"
>       src="xxxx.png"
>       @click.stop.prevent="$emit('open')"
>     />
>   </component>
> ```
> 
> 我使用usedraggable组件，当我释放鼠标时，也触发了click方法，但我已经设置了： `：stopPropagation=“true”`； 我读了源代码，可能是以下问题造成的：
> 
> I use the usedraggable component， when I release the mouse, triggered the click method too, but i already set : `:stopPropagation="true"`; I read the source code, which may be caused by the following problems：
> 
> image

请问你是怎么引入的 Draggable组件的  我这样引入不对
WeChat6f20e7ccf8c5289bde7c7bdf5b9ed8c4

**Comment by DamonCais** [comment]
It seems that the stopPropagation option was not working.

https://play.vueuse.org/#N4IgDghgxg1hDmBTAziAXAbVAOwgW0XRADcBXRAWgBNE8BLEAGhGQHtSAnKQtEU7MDHhMQNZFA50wAFzqtsRAGrkABABFadFQAoAFhAA2AMxV1sKgGIdE2KLoCUp5CogqaxRAdZgz8FaVkDOmkATxcDLwB3ZxD2FWlWFUjJaUQVAFVsOg8OZEMVZTSAGToAIw4ISRQVI1YOAtUAJhUAMhUAZhFOAyJdaWkwZDQAemH+QXgAOihWPGGyShp6YaDS4bMaAA9JvAArVABfRhx8HhAAAQXSZERh5H1rKhE2Tm4icaERMQkpWXkiADKD0QVAaiHSN38gWCdBQky6HB6vD6AyGow+Uxmc0u5Gut3ulRB5wAjAAGSbEyak9bYLY7fYgI4nAhEHGIPHDGbWZ7sLhnDFfFA-GRyBS8ADCrAiiCgf3MrBMKBu2Fk+UKKkleDArGQMPkGWhshQCKRIBRgxGYwEQmms2GbI5XMQJPJlOpG0Q2z2hwAuswjHQDMa0KAAIJgMCTBboUCpLUGCCpIgAHio2QAOuYVPBJFQALzpkAAPxqUQoMwM2YgYAoABZC5mVCowAnuAWQDMVTZpOXu4h6sFaMhe12OA2s7p2+JrDZx03air23h5Kw5yoAHyZxsqVPZFTWIztzyFlRoZChINTi+IE-nkKXwva3VytA1OibEGFzdZpu74jfpsmzUCo-AIABCFQAEkXDwFxpBUYBgE2A4jgQ4AQhQrcf0A5NSgCBJzHOKAglgdtUnPL9yOkZNhjw-p5AAnDhjTf9tybbcaJY78sM47J12eYVpCIOgtTqeDgH8G4AFl2EhA4ag4WYVAAcgdG5OTqRBlKwkTtQ4cT90QEx5KMRTYOUhZtKyUT9IQyTEGAhB4AgUogxUEyzJUtTbidKzM07c8VE8FQ80MoxtGwUgInsLCAoMzZGBUEJErvNz5NCvFHPgZzXMQbRPES4BtzMGFDEUQxyFfCTNlfWtSUSkJatJdzGG3c9vAABUUyBnJfaQOHITMDhi7B-PkQKqJCnREHsPN1yKrMAqlRBJi8eB8vsABuIbGWYSU6jTXBUijcgYxAOMW0TM4-23BdpCXFc11KeB2yCeA+goABWUlmqoSoYDQP6OBgL6frXIwg02dsSwrNdUk2e7C24Uc1zAdtGieg7+3bRT+BoKhx0Y5NkEgcx4cRkBa02Awv0QlRiAqtIUJokmIGwInWbJz0KeQWCgYBt6PoATh+tx-rQbxoGCEJQZPPAyIoDGQHXGS8TQlQE1KYLmbuUnv14-8BMkGQiBoANsEQLrvGQbQFqbTXPFfAF+t8VqswZgxKpUAA5Ug8C1jg3eGxkUKAA

**Comment by wesuredata** [comment]
### 在图片上面覆盖一个空的元素
```vue
<div  ref="cameraRef" :style="[style, { position: 'fixed' }]">
    <!-- 外面包一层，控制遮罩层 -->
    <div style="position: relative;">
        <p>拖拽文本没问题, 拖拽下面图片就会被图片的默认事件干扰</p>
        <img class="camera_icon" src="@/views/preview/assets/xxxx/camera_drag.png" />
        <!-- 下面就是一个虚拟的遮罩层 -->
        <div style="position: absolute;inset: 0;z-index: 100;"></div>
    </div>
</div>

<script>
const { style } = useDraggable(cameraRef, {
  initialValue: { x: 500, y: 500 },
})
</script>
```

**Pull request merged**